modules/Bio/EnsEMBL/Compara/RunnableDB/UpdateMSA/UpdateGAB.pm

```diff
@@ -0,0 +1,279 @@
+            # If only one child is removed, the other child will replace this node in the GAT
+            if ( (scalar $children_to_rm{$inner_node->dbID} == 2) && defined $inner_node->parent ) {
+                push @{ $children_to_rm{$inner_node->parent->dbID} }, $inner_node->dbID;
+            }
+        }
+        # Each inner node in a GAT must have two children: if at least one is removed, the inner node has to
+        # be removed as well
+        $node_ids_to_rm{$_} = 1 for keys %children_to_rm;
+        # Get the list of ancestral GAs, dnafrags and nodes to remove, plus the nodes to relocate in the GAT
+        foreach my $node ( @{ $gat_root->get_all_nodes() } ) {
+            my $node_ga = $node->get_all_genomic_aligns_for_node->[0];
+            if ( exists $node_ids_to_rm{$node->dbID} ) {
+                $ga_ids_to_rm{$node_ga->dbID} = 1;
+                push @dnafrag_ids_to_rm, $node_ga->dnafrag_id unless $node->is_leaf();
+            } else {
+                if ( $node->has_parent() && exists $node_ids_to_rm{$node->parent->dbID} ) {
+                    # Find the final parent node, as it can be a (great-)*grandparent
+                    my $new_parent = $node->parent->parent;
+                    while ( defined $new_parent && exists $node_ids_to_rm{$new_parent->dbID} ) {
+                        $new_parent = $new_parent->parent;
+                    }
+                    # If $parent is undefined we have reached the GAT root, so this node will be the new root
+                    $gat_root = $node unless defined $new_parent;
+                    $node_ids_to_update{$node->dbID} = {
+                        parent_id    => defined $new_parent ? $new_parent->dbID : undef,
+                        prev_root_id => $node->root->dbID,  # only used to update root_id
+                        distance     => defined $new_parent ? $node->distance_to_node($new_parent) : $node->distance_to_root(),
+                    };
+                }
+                # Also include the ancestral GAs that will remain to update their CIGAR lines
+                push @$genomic_aligns, $node_ga unless $node->is_leaf();
+            }
+        }
+        # The ancestral GAB will need to be updated as well
+        push @gabs_to_update, $genomic_aligns->[-1]->genomic_align_block;
+    }
+    # Get the new CIGAR lines, discarding any gap-only column produced by the removal of the marked GAs
+    my @prev_cigars = map { $_->cigar_line } @$genomic_aligns;
+    my @updated_cigars = Bio::EnsEMBL::Compara::Utils::Cigars::minimize_cigars(@prev_cigars);
+    my %gas_to_update;
+    while ( my ($index, $cigar) = each @updated_cigars ) {
+        next if $cigar eq $prev_cigars[$index];
+        $gas_to_update{$genomic_aligns->[$index]->dbID} = $cigar;
+    }
+    # Get the new alignment length
+    my $new_aln_length = Bio::EnsEMBL::Compara::Utils::Cigars::alignment_length_from_cigar($updated_cigars[0]);
```

Comment:
I feel like removing gap only columns would be a useful method to have on the GenomicAlignBlock object itself (and it seems like the logical place to keep this sort of functionality)

Reply:
To be added to the pipeline's 2.0 version ;)